manifests/init.pp

```diff
@@ -321,14 +326,19 @@
   validate_bool($force_online)
   validate_bool($check_standby)
   validate_bool($log_file)
-  if getvar('log_file_name') and $log_file == true {
-    validate_absolute_path($log_file_name)
-  }
-  validate_bool($log_timestamp)
   validate_bool($debug)
   validate_bool($log_stderr)
   validate_re($syslog_priority, '^(debug|info|notice|warning|err|emerg)$')
   validate_bool($log_function_name)
+  if $log_subsystem {
+    [$log_subsystem].each |Hash $subsystem| {
+      validate_string($subsystem[name])
+      validate_bool($subsystem[debug])
+      validate_bool($subsystem[log_file])
+      validate_bool($subsystem[syslog])
```

Comment:
Since this is Puppet >4.x already, you should use data types rather than `validate_xx`, and probably use a flatter structure for configuration of the params? Especially as implemented / documented here, it seems like it would be hard for someone to configure this without reading the code.